### PR TITLE
RELATED: FET-1113 Enzyme to RTL - ui-kit - (button, chartSorting, colorPicker)

### PR DESCRIPTION
**Pull request by marmil-cz**
---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9335/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11167/ : FAILURE in 44s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9336/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11168/ : FAILURE in 10m 05s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9349/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11184/ : FAILURE in 10m 15s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9350/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11187/ : FAILURE in 19m 30s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9364/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11206/ : FAILURE in 1m 03s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by marmil-cz** [comment]
extended test

**Comment by marmil-cz** [comment]
extended check cypress

File: libs/sdk-ui-kit/package.json [hunk 1]
```diff
@@ -154,6 +154,10 @@
         "stylelint": "^13.8.0",
         "svgo": "^2.8.0",
         "ts-jest": "^27.0.5",
-        "typescript": "4.0.2"
+        "typescript": "4.0.2",
+        "@testing-library/react": "^12.1.5",
```
Comment (NestorEncinas) [inline libs/sdk-ui-kit/package.json:158]: Please keep them in alphabetical order.

File: libs/sdk-ui-kit/src/Button/tests/Button.test.tsx [hunk 2]
```diff
@@ -1,95 +1,98 @@
...
+            await waitFor(() => {
+                expect(onClick).not.toHaveBeenCalled();
+            });
         });
     });
 
     describe("selenium class", () => {
         it("should have correct selenium class", () => {
-            const wrapper = renderButton({
+            renderButton({
```
Comment (NestorEncinas) [inline libs/sdk-ui-kit/src/Button/tests/Button.test.tsx:46]: Perhaps we can get rid of this test block. I don't think it has value enough to be tested. What do you think?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9373/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11215/ : FAILURE in 10m 50s

File: libs/sdk-ui-kit/src/Button/tests/Button.test.tsx (hunk 2 above)
Comment (marmil-cz) [reply libs/sdk-ui-kit/src/Button/tests/Button.test.tsx:46]: makes sense

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**headlineChartDrilling.spec.ts**|2|-|8.9110s|
|✅|**barChartDrilling.spec.ts**|4|-|11.5010s|
|✅|**pivotTableAggregationsMenuAndAddTotal.spec.ts**|5|-|25.8280s|
|✅|**pivotTableSizingReset.spec.ts**|15|-|44.0090s|
|✅|**pivotTableSizing.spec.ts**|1|-|9.7850s|
|✅|**attributeFilterButtonWithSelection.spec.ts**|2|-|11.6770s|
|✅|**drilling.spec.ts**|2|-|21.7680s|
|✅|**attributeFilterConfig.spec.ts**|3|-|31.2350s|
|✅|**pivotTableRemoveAllFromMutipleAggregation.spec.ts**|1|-|11.9980s|
|✅|**attributeFilterButtonParentChild.spec.ts**|3|-|19.4380s|
|✅|**pivotTableRemoveOneAggregation.spec.ts**|1|-|12.0150s|
|✅|**parentChildFiltersWithPlaceholders.spec.ts**|3|-|23.6940s|
|✅|**dashboardFiltering.spec.ts**|2|-|27.2860s|
|✅|**pivotTableRemoveOneFromMutipleAggregation.spec.ts**|1|-|12.1840s|
|✅|**attributeFilterButton.spec.ts**|9|-|36.1620s|
|✅|**dashboard.spec.ts**|9|-|55.0960s|
|✅|**dateFilter.spec.ts**|9|-|20.4490s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/282/).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9375/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/282/ : SUCCESS in 15m 12s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1798/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9374/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1798/ : SUCCESS in 26m 11s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9377/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11217/ : FAILURE in 19m 45s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9380/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11220/ : FAILURE in 20m 21s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9388/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11230/ : SUCCESS in 20m 15s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9392/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11234/ : SUCCESS in 19m 32s

**Review by NestorEncinas — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2867

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2653/ : SUCCESS in 1m 48s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2954/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1787/ : SUCCESS in 15m 32s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11243/ : SUCCESS in 19m 46s

**Pull request merged**